display_list/dl_color.h

```diff
@@ -58,58 +77,111 @@ struct DlColor {
+  constexpr int getAlpha() const { return toC(alpha_); }
+  ///\deprecated Use floating point accessors to avoid data loss when using wide
+  /// gamut colors.
+  constexpr int getRed() const { return toC(red_); }
+  ///\deprecated Use floating point accessors to avoid data loss when using wide
+  /// gamut colors.
+  constexpr int getGreen() const { return toC(green_); }
+  ///\deprecated Use floating point accessors to avoid data loss when using wide
+  /// gamut colors.
+  constexpr int getBlue() const { return toC(blue_); }
+
+  constexpr DlScalar getAlphaF() const { return alpha_; }
+  constexpr DlScalar getRedF() const { return red_; }
+  constexpr DlScalar getGreenF() const { return green_; }
+  constexpr DlScalar getBlueF() const { return blue_; }
+
+  constexpr DlColorSpace getColorSpace() const { return color_space_; }
+
+  ///\deprecated Use floating point accessors to avoid data loss when using wide
+  /// gamut colors.
   constexpr uint32_t premultipliedArgb() const {
     if (isOpaque()) {
-      return argb_;
+      return argb();
     }
     DlScalar f = getAlphaF();
-    return (argb_ & 0xFF000000) |       //
+    return (argb() & 0xFF000000) |      //
            toC(getRedF() * f) << 16 |   //
            toC(getGreenF() * f) << 8 |  //
            toC(getBlueF() * f);
   }
 
   constexpr DlColor withAlpha(uint8_t alpha) const {  //
-    return DlColor((argb_ & 0x00FFFFFF) | (alpha << 24));
+    return DlColor((argb() & 0x00FFFFFF) | (alpha << 24));
   }
   constexpr DlColor withRed(uint8_t red) const {  //
-    return DlColor((argb_ & 0xFF00FFFF) | (red << 16));
+    return DlColor((argb() & 0xFF00FFFF) | (red << 16));
   }
   constexpr DlColor withGreen(uint8_t green) const {  //
-    return DlColor((argb_ & 0xFFFF00FF) | (green << 8));
+    return DlColor((argb() & 0xFFFF00FF) | (green << 8));
   }
   constexpr DlColor withBlue(uint8_t blue) const {  //
-    return DlColor((argb_ & 0xFFFFFF00) | (blue << 0));
+    return DlColor((argb() & 0xFFFFFF00) | (blue << 0));
+  }
+  constexpr DlColor withAlphaF(float alpha) const {  //
+    return DlColor(alpha, red_, green_, blue_, color_space_);
+  }
+  constexpr DlColor withRedF(float red) const {  //
+    return DlColor(alpha_, red, green_, blue_, color_space_);
+  }
+  constexpr DlColor withGreenF(float green) const {  //
+    return DlColor(alpha_, red_, green, blue_, color_space_);
   }
+  constexpr DlColor withBlueF(float blue) const {  //
+    return DlColor(alpha_, red_, green_, blue, color_space_);
+  }
+  DlColor withColorSpace(DlColorSpace color_space) const;
 
   constexpr DlColor modulateOpacity(DlScalar opacity) const {
     return opacity <= 0   ? withAlpha(0)
            : opacity >= 1 ? *this
                           : withAlpha(round(getAlpha() * opacity));
   }
 
-  constexpr uint32_t argb() const { return argb_; }
+  ///\deprecated Use floating point accessors to avoid data loss when using wide
+  /// gamut colors.
+  constexpr uint32_t argb() const {
```

Comment:
Should this adjust to sRGB?

Reply:
Maybe a fast path if it is already sRGB?

Reply:
No, it still calculates sensible output.  Wether it is what we want is asserted at the integration test level.